homeassistant/components/openai_conversation/conversation.py

```diff
@@ -95,14 +97,23 @@ def _message_convert(message: ChatCompletionMessage) -> ChatCompletionMessagePar
 def _chat_message_convert(
     message: conversation.Content
     | conversation.NativeContent[ChatCompletionMessageParam],
+    model: str,
 ) -> ChatCompletionMessageParam:
     """Convert any native chat message for this agent to the native format."""
-    if message.role == "native":
+    role = message.role
+    if role == "native":
         # mypy doesn't understand that checking role ensures content type
         return message.content  # type: ignore[return-value]
+    if role == "system":
+        # Early reasoning models don't support system messages
+        if model.startswith(("o1-mini", "o1-preview")):
```

Comment:
Given that we need so many gotchas for these 2 models, what about we don't allow configuring them to begin with? o3-mini is priced the same as o1-mini.

Reply:
Done. It does simplify a few things.